railties/lib/rails/generators/rails/app/templates/config/boot.rb

```diff
@@ -1,3 +1,17 @@
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
 
 require 'bundler/setup' # Set up gems listed in the Gemfile.
+
+if RUBY_ENGINE == 'ruby' && RUBY_PLATFORM =~ /darwin|linux|bsd/
+  require 'bootsnap'
+
+  Bootsnap.setup(
+    cache_dir:            File.expand_path('../tmp/cache', __dir__),
+    development_mode:     ['development', '', nil].include?(ENV['RAILS_ENV']),
+    disable_trace:        false,
+    load_path_cache:      true,
+    autoload_paths_cache: true,
+    compile_cache_iseq:   true,
+    compile_cache_yaml:   true
```

Comment:
Could also be `require "bootsnap/railtie"` or Rails could just set up bootsnap automatically after executing `boot.rb` if the gem is present, e.g.:

```ruby
# Somewhere in the Rails bowels:
begin
  require "bootsnap"

  Bootsnap.setup(…)
rescue LoadError
end
```